### PR TITLE
Improve idlharness-ShadowRealm integration

### PR DESCRIPTION
**Pull request by ptomato**
By defining `globalThis.fetch_spec()` in `testharness-shadowrealm-inner.js`, we no longer need to have IDL tests in separate files for ShadowRealm, as the regular `idlharness.js` will now just run there.

This allows consolidating the `idlharness-shadowrealm.window.js` tests back into their neighbouring `idlharness.any.js` files.

### CODE REVIEW

File: resources/idlharness.js [hunk 1]
```diff
@@ -1422,7 +1422,7 @@ IdlInterface.prototype.test = function()
         {
             subsetTestByKey(this.name, test, function() {
                 assert_false(this.name in self);
-            }.bind(this), this.name + " interface: existence and properties of interface object");
+            }.bind(this), this.name + " interface: nonexistence of interface object");
```
Comment (Ms2ger) [inline resources/idlharness.js:1425]: Hmm, I guess the reason for this is that we then always have a subtest with this name. Would it be sufficient to add a message to the `assert_false`?

File: resources/idlharness.js [hunk 2]
```diff
@@ -3451,6 +3451,17 @@ IdlNamespace.prototype.test_self = function ()
 
 IdlNamespace.prototype.test = function ()
 {
+    // If the namespace object is not exposed, only test that. Members can't be
```
Comment (Ms2ger) [inline resources/idlharness.js:3454]: This broke the `tests/functional/idlharness/IdlNamespace/test_operation.html` unit test, I think

Comment (ptomato) [reply resources/idlharness.js:3454]: Fixed and added an additional test for the new functionality.

**Review by inexorabletash — APPROVED** [review]
LGTM, for what it's worth.

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**